zone/gm_commands/time.cpp

```diff
@@ -2,31 +2,144 @@
 
 void command_time(Client *c, const Seperator *sep)
 {
-	char timeMessage[255];
-	int  minutes = 0;
+	int arguments = sep->argnum;
+	if (!arguments || !sep->IsNumber(1)) {
+		c->Message(Chat::White, "Usage: #time [Hour] [Minute]");
+
+		TimeOfDay_Struct world_time;
+		zone->zone_time.GetCurrentEQTimeOfDay(time(0), &world_time);
+		
+		auto time_string = fmt::format(
+			"{}:{}{} {} (Timezone: {}:{}{} {})",
+			(
+				((world_time.hour - 1) % 12) == 0 ?
+				12 :
+				((world_time.hour - 1) % 12)
+			),
+			(
+				world_time.minute < 10 ?
+				"0" :
+				""
+			),
+			world_time.minute,
+			(
+				world_time.hour >= 13 ?
+				"PM" :
+				"AM"
+			),
+			(
+				((zone->zone_time.getEQTimeZoneHr() - 1) % 12) == 0 ?
+				12 :
+				((zone->zone_time.getEQTimeZoneHr() - 1) % 12)
+			),
+			(
+				zone->zone_time.getEQTimeZoneMin() < 10 ?
+				"0" :
+				""
+			),
+			zone->zone_time.getEQTimeZoneMin(),
+			(
+				zone->zone_time.getEQTimeZoneHr() >= 13 ?
+				"PM" :
+				"AM"
+			)
+		);
+
+		c->Message(
+			Chat::White,
+			fmt::format(
+				"It is currently {}.",
+				time_string
+			).c_str()
+		);
+
+		return;
+	}
+
 	if (sep->IsNumber(1)) {
+		uint8 minutes = 0;
+		auto hours = static_cast<uint8>(std::stoul(sep->arg[1]) + 1);
 		if (sep->IsNumber(2)) {
-			minutes = atoi(sep->arg[2]);
+			minutes = static_cast<uint8>(std::stoul(sep->arg[2]));
 		}
-		c->Message(Chat::Red, "Setting world time to %s:%i (Timezone: 0)...", sep->arg[1], minutes);
-		zone->SetTime(atoi(sep->arg[1]) + 1, minutes);
-		LogInfo("{} :: Setting world time to {}:{} (Timezone: 0)...", c->GetCleanName(), sep->arg[1], minutes);
-	}
-	else {
-		c->Message(Chat::Red, "To set the Time: #time HH [MM]");
-		TimeOfDay_Struct eqTime;
-		zone->zone_time.GetCurrentEQTimeOfDay(time(0), &eqTime);
-		sprintf(
-			timeMessage, "%02d:%s%d %s (Timezone: %ih %im)",
-			((eqTime.hour - 1) % 12) == 0 ? 12 : ((eqTime.hour - 1) % 12),
-			(eqTime.minute < 10) ? "0" : "",
-			eqTime.minute,
-			(eqTime.hour >= 13) ? "pm" : "am",
-			zone->zone_time.getEQTimeZoneHr(),
-			zone->zone_time.getEQTimeZoneMin()
+
+		c->Message(
+			Chat::White,
+			fmt::format(
+				"Setting world time to {}:{}{} {} (Timezone: {}:{}{} {}).",
+				(
+					((hours - 1) % 12) == 0 ?
+					12 :
+					((hours - 1) % 12)
+				),
+				(
+					minutes < 10 ?
+					"0" :
```

Comment:
Might be a little cleaner to pad the `0` to read `00:00` vs `0:0` like clock time

Example: https://github.com/EQEmu/Server/blob/master/common/server_event_scheduler.cpp#L36

Also I see this ternary is used in a lot of places in this PR; might be worth extracting to a helper method under the `EQTime` class to read back the time in string form